v3ioutils/container.go

```diff
@@ -155,48 +162,50 @@ func DeleteTable(logger logger.Logger, container v3io.Container, path, filter st
 	return nil
 }
 
-func respWaitLoop(logger logger.Logger, comm chan int, responseChan chan *v3io.Response, timeout time.Duration) chan bool {
-	responses := 0
-	requests := -1
-	done := make(chan bool)
-
-	go func() {
-		active := false
-		for {
-			select {
-
-			case resp := <-responseChan:
-				responses++
-				active = true
-
-				if resp.Error != nil {
-					logger.ErrorWith("failed Delete response", "error", resp.Error)
-					// TODO: signal done and return?
-				}
-
-				if requests == responses {
-					done <- true
-					return
-				}
-
-			case requests = <-comm:
-				if requests <= responses {
-					done <- true
-					return
-				}
+func getItemsWorker(container v3io.Container, input *v3io.GetItemsInput, fileNameChan chan<- string, terminationChan chan<- error, onErrorTerminationChannel <-chan struct{}) {
+	for {
+		select {
+		case _ = <-onErrorTerminationChannel:
+			terminationChan <- nil
+			return
+		default:
+		}
+		resp, err := container.GetItemsSync(input)
+		if err != nil {
+			terminationChan <- err
+			return
+		}
+		resp.Release()
+		output := resp.Output.(*v3io.GetItemsOutput)
+		for _, item := range output.Items {
+			fileNameChan <- item.GetField("__name").(string)
+		}
+		if output.Last {
+			terminationChan <- nil
+			return
+		}
+		input.Marker = output.NextMarker
+	}
+}
 
-			case <-time.After(timeout):
-				if !active {
-					logger.ErrorWith("Resp loop timed out!", "requests", requests, "response", responses)
-					done <- true
-					return
-				}
-				active = false
+func deleteObjectWorker(tablePath string, container v3io.Container, fileNameChan <-chan string, outgoingTerminationChan chan<- error, onErrorTerminationChannel <-chan struct{}) {
```

Comment:
i would change `outgoingTerminationChan` to just `terminationChan` so it would be the same as the getItemsWorker for clarity